src/kaocha/config.clj

```diff
@@ -217,6 +222,39 @@
                           tests)))))
     config))
 
+(defn load-config2
```

Comment:
Let's pick something descriptive, this does quite a bit now so we have several options:

- `load-config-with-checks`
- `load-cli-config`
- `load-config-for-profile`
- `load-config-and-validate`

Looking at it this way I think we can factor this much better as a multi-stage thing

I think we can keep profile in `opts`, no need to introduce an extra arity for that. But maybe make `read-config-source` deal more gracefully with `{:profile nil}`, using the same fallback as when the profile is absent. That would also simplify the calling code. (you can then do `{:profile profile}` instead of `(if profile {:profile profile} {})`

```clj
(defn apply-cli [config cli-opts cli-args]
  (-> config
      (apply-cli-opts cli-opts)
      (apply-cli-args cli-args)))

(defn validate! [config config-file]
  ;; print warning
  ;; check spec, might throw
  )

(defn load-config-for-cli-and-validate [config-file opts]
  (-> config-file
      (load-config opts)
      (apply-cli cli-opts cli-args)
      (validate! config-file)))
```